docs/tutorial/procedure.rst

```diff
@@ -249,6 +249,52 @@ Let's extend our SimpleProcedure with logging. ::
 First, we have imported the Python logging module and grabbed the logger using the :python:`__name__` argument. This gives us logging information specific to the current file. Conversely, we could use the :python:`''` argument to get all logs, including those of pymeasure. We use the :python:`console_log` function to conveniently output the log to the console. Further details on how to use the logger are addressed in the Python logging documentation.
 
 
+Storing metadata
+~~~~~~~~~~~~~~~~
+
+Metadata (:class:`pymeasure.experiment.parameters.Metadata`) allows storing information (e.g. the actual starting time, instrument parameters) about the measurement in the header of the datafile.
+These Metadata objects are evaluated and stored in the datafile only after the :python:`startup` method has ran; this way it is possible to e.g. retrieve settings from an instrument and store them in the file.
+Using a Metadata is nearly as straightforward as using a Parameter; extending the example of above to include metadata, looks as follows: ::
+
+    from time import sleep, time
+    from pymeasure.experiment import Procedure
+    from pymeasure.experiment import IntegerParameter, Metadata
+
+    class SimpleProcedure(Procedure):
+
+        # a Parameter that defines the number of loop iterations
+        iterations = IntegerParameter('Loop Iterations')
+
+        # the Metadata objects store information after the startup has ran
+        starttime = Metadata('Start time', fget=time)
+        custom_metadata = Metadata('Custom', default=1)
+
+        # a list defining the order and appearance of columns in our data file
+        DATA_COLUMNS = ['Iteration']
+
+        def startup(self):
+            self.custom_metadata = 20
+
+        def execute(self):
+            """ Loops over each iteration and emits the current iteration,
+            before waiting for 0.01 sec, and then checking if the procedure
+            should stop
+            """
+            for i in range(self.iterations):
+                self.emit('results', {'Iteration': i})
+                sleep(0.01)
+                if self.should_stop():
+                    break
+
+
+As with a Parameter, PyMeasure swaps out the Metadata with their values behind the scene, which makes accessing the values of Metadata very convenient.
+
+The value of a Metadata can be set either using an :python:`fget` method or manually in the startup method.
+The :python:`fget` method, if provided, is ran after startup method.
+It can also be provided as a string; in that case it is assumed that the string contains the name of a method of the Procedure class which returns the value that is to be stored.
```

Comment:
Why this duplicated way of defining the callback? Can't you just say `Metadata('bla', fget=self.mymethod)` instead of `Metadata('bla', fget='mymethod')`? 
The only limitation, I think, is that the Metadata class attribute has to be defined _after_ that method, but that kinda makes sense anyway. 
Additionally, why have the Metadata as class-level attributes? Would it not be simpler to just define instance attributes in the __init__? That would also avoid the above limitation.

Reply:
The reason is exactly as you mention, the fact that you can only define it after defining the the method. Why I chose for this (and also to make it a class attribute) is to make it similar to the Parameter and Measurable objects, such that you can define this at the beginning of the Parameters class: everything that is stored in the file can be nicely defined at a single place, making it (in my opinion) neater/clearer.